204 No Content improvements

For example,

```go
func (c *Controller) Create() {
  // do something
}
```

These functions should return `204 No Content` if successful.

> Okay, so I think it's a bit more nuanced then. I saw an error when you only have one controller action and that controller action had no results. I think it's an "import is not used" error. If so, should be pretty easy to fix, just moving an `imports.AddNamed` to a different location in the controller loader.

So I did a bit of playing around with different void scenarios, and the only scenario I found that didn't work was having a void action tied to a view. It resulted in:
```
# github.com/go-duo/bud/example/basic/bud/.app/controller
bud/.app/controller/controller.go:50:24: not enough arguments in call to i.View.Handler
        have (string)
        want (string, interface{})
```

> Yep, I totally think it makes sense for `func (c *Controller) Action(...) error { return nil }` to also return a `204 No Content`

Both of these are fixed in #68

--- a/runtime/generator/controller/controller.gotext
+++ b/runtime/generator/controller/controller.gotext
@@ -118,6 +118,8 @@ func ({{$action.Short}} *{{ $.Pascal }}{{$action.Pascal}}Action) handler(httpReq
 		{{- if $action.RespondJSON }}
 		{{- if $action.Results.Result }}
 		JSON: response.JSON({{ $action.Results.Result }}),
+		{{- else if $action.Results.IsOnlyError }}
+		JSON: response.Status(204),
 		{{- else }}
 		JSON: response.Status(200).Set("Content-Type", "application/json"),
 		{{- end }}

--- a/runtime/generator/controller/controller_test.go
+++ b/runtime/generator/controller/controller_test.go
@@ -1895,11 +1895,11 @@ func TestWorkingChangeWorking(t *testing.T) {
 }
 
 func TestEmptyActionWithView(t *testing.T) {
-	t.SkipNow()
 	is := is.New(t)
 	ctx := context.Background()
 	dir := t.TempDir()
 	bud := budtest.New(dir)
+	bud.NodeModules["svelte"] = version.Svelte
 	bud.Files["controller/controller.go"] = `
 		package controller
 		type Controller struct {}
@@ -1911,6 +1911,7 @@ func TestEmptyActionWithView(t *testing.T) {
 	app, err := project.Build(ctx)
 	is.NoErr(err)
 	is.NoErr(app.Exists("bud/.app/view/view.go"))
+	is.NoErr(app.Exists("bud/.app/controller/controller.go"))
 	is.NoErr(app.Exists("bud/.app/main.go"))
 	server, err := app.Start(ctx)
 	is.NoErr(err)

--- a/runtime/generator/controller/controller_test.go
+++ b/runtime/generator/controller/controller_test.go
@@ -951,15 +951,14 @@ func TestNestedResource(t *testing.T) {
 	res, err = server.PatchJSON("/users/10", bytes.NewBufferString(`{"name": "matt", "age": 10}`))
 	is.NoErr(err)
 	is.NoErr(res.Expect(`
-		HTTP/1.1 200 OK
-		Content-Type: application/json
+		HTTP/1.1 204 No Content
+		Content-Length: 0
 		Date: Fri, 31 Dec 2021 00:00:00 GMT
 	`))
 	res, err = server.DeleteJSON("/users/10", nil)
 	is.NoErr(err)
 	is.NoErr(res.Expect(`
-		HTTP/1.1 200 OK
-		Content-Type: application/json
+		HTTP/1.1 204 No Content
 		Date: Fri, 31 Dec 2021 00:00:00 GMT
 	`))
 }

--- a/runtime/generator/controller/state.go
+++ b/runtime/generator/controller/state.go
@@ -199,6 +199,11 @@ func (results ActionResults) Error() string {
 	return ""
 }
 
+// Error expression is only return
+func (results ActionResults) IsOnlyError() bool {
+	return len(results) == 1 && results[0].IsError
+}
+
 // ActionResult struct
 type ActionResult struct {
 	Name     string

--- a/runtime/generator/controller/state.go
+++ b/runtime/generator/controller/state.go
@@ -159,14 +159,13 @@ func isList(dataType string) bool {
 
 func (results ActionResults) ViewResult() string {
 	propsKey := results.propsKey()
-	if propsKey == "" {
-		return results.Result()
-	}
 	out := new(strings.Builder)
 	out.WriteString(`map[string]interface{}{`)
-	out.WriteString(strconv.Quote(propsKey))
-	out.WriteString(":")
-	out.WriteString(results.Result())
+	if propsKey != "" {
+		out.WriteString(strconv.Quote(propsKey))
+		out.WriteString(":")
+		out.WriteString(results.Result())
+	}
 	out.WriteString(`},`)
 	return out.String()
 }

